oxidation/libparsec/crates/libparsec_fs/src/file_operations.rs

```diff
@@ -0,0 +1,603 @@
+        assert_eq!(chunk0.raw_offset, 0);
+        assert_eq!(chunk0.raw_size.get(), 6);
+        assert_eq!(chunk0.access, None);
+
+        // Append more data to the file and read everything back
+        let t3 = DateTime::from_str("2000-01-01 03:00:00 UTC").unwrap();
+        manifest = storage.write(&manifest, b"world !", 6, t3);
+        assert_eq!(storage.read(&manifest, 13, 0), b"Hello world !");
+    }
+}
+
+// ...
+//
+// (_, chunk1), = manifest.blocks
+// assert manifest == base.evolve(size=13, blocks=((chunk0, chunk1),), updated=t3)
+// assert chunk1 == Chunk(id=chunk1.id, start=6, stop=13, raw_offset=6, raw_size=7, access=None)
+// assert storage[chunk1.id] == b"world !"
+//
+// with freeze_time("2000-01-04") as t4:
+// manifest = storage.write(manifest, b"\n More kontent", 13, t4)
+// assert storage.read(manifest, 27, 0) == b"Hello world !\n More kontent"
+//
+// (_, _, chunk2), (chunk3,) = manifest.blocks
+// assert storage[chunk2.id] == b"\n M"
+// assert storage[chunk3.id] == b"ore kontent"
+// assert manifest == base.evolve(
+// size=27, blocks=((chunk0, chunk1, chunk2), (chunk3,)), updated=t4
+// )
+//
+// with freeze_time("2000-01-05") as t5:
+// manifest = storage.write(manifest, b"c", 20, t5)
+// assert storage.read(manifest, 27, 0) == b"Hello world !\n More content"
+//
+// chunk4, chunk5, chunk6 = manifest.blocks[1]
+// assert chunk3.id == chunk4.id == chunk6.id
+// assert storage[chunk5.id] == b"c"
+// assert manifest == base.evolve(
+// size=27, blocks=((chunk0, chunk1, chunk2), (chunk4, chunk5, chunk6)), updated=t5
+// )
+//
+// with freeze_time("2000-01-06") as t6:
+// manifest = storage.resize(manifest, 40, t6)
+// expected = b"Hello world !\n More content" + b"\x00" * 13
+// assert storage.read(manifest, 40, 0) == expected
+//
+// (_, _, _, chunk7), (chunk8,) = manifest.blocks[1:]
+// assert storage[chunk7.id] == b"\x00" * 5
+// assert storage[chunk8.id] == b"\x00" * 8
+// assert manifest == base.evolve(
+// size=40,
+// blocks=((chunk0, chunk1, chunk2), (chunk4, chunk5, chunk6, chunk7), (chunk8,)),
+// updated=t6,
+// )
+//
+// with freeze_time("2000-01-07") as t7:
+// manifest = storage.resize(manifest, 25, t7)
+// expected = b"Hello world !\n More conte"
+// assert storage.read(manifest, 25, 0) == expected
+//
+// (_, _, chunk9), = manifest.blocks[1:]
+// assert chunk9.id == chunk6.id
+// assert manifest == base.evolve(
+// size=25, blocks=((chunk0, chunk1, chunk2), (chunk4, chunk5, chunk9)), updated=t7
+// )
+//
+// with freeze_time("2000-01-08"):
+// assert not manifest.is_reshaped()
+// manifest = storage.reshape(manifest)
+// expected = b"Hello world !\n More conte"
+// assert storage.read(manifest, 25, 0) == expected
+// assert manifest.is_reshaped()
+//
+// (chunk10,), (chunk11,) = manifest.blocks
+// assert storage[chunk10.id] == b"Hello world !\n M"
+// assert storage[chunk11.id] == b"ore conte"
+// assert manifest == base.evolve(size=25, blocks=((chunk10,), (chunk11,)), updated=t7)
+//
```

Comment:
What's the use for those comments ?

Reply:
They were temporary, I was migrating this test